bemserver_ui/static/scripts/modules/views/analysis/weather.js

```diff
@@ -0,0 +1,330 @@
+import { InternalAPIRequest } from "/static/scripts/modules/tools/fetcher.js";
+import { FlashMessageTypes, FlashMessage } from "/static/scripts/modules/components/flash.js";
+import { flaskES6, signedUser } from "/static/scripts/app.js";
+import { Spinner } from "/static/scripts/modules/components/spinner.js";
+import { TimeseriesChartWeather} from "/static/scripts/modules/components/charts/tsChartWeather.js";
+import "/static/scripts/modules/components/tree.js";
+import { TimeCalendar } from "/static/scripts/modules/tools/time.js";
+
+export class WeatherExploreView {
+
+    #internalAPIRequester = null;
+    #retrieveDataReqID = null;
+    #sitesTreeReqID = null;
+
+    #messagesElmt = null;
+    #mainChartContainerElmt = null;
+    #periodTypeSelectElmt = null;
+    #periodDaySelectElmt = null;
+    #periodMonthSelectElmt = null;
+    #periodYearSelectElmt = null;
+    #sitesTreeElmt = null;
+
+    #forecast = null;
+    #forecastWrapper = null;
+
+    #tzName = "UTC";
+    #yearRef = null;
+    #monthRef = null;
+    #dayRef = null;
+    #maxPastYears = 20;
+
+    #structuralElementType = null;
+    #structuralElementId = null;
+    #chartWeather = {};
+
+    #previousPeriodType = null;
+    #previousDaySelected = null;
+    #previousYearSelected = null;
+
+    #timeFormatPerPeriodType = {
+        "Day-Minute": "{dd} {MMMM} {yyyy} {HH}:{mm}",
+        "Week-Hourly": "{dd} {MMMM} {yyyy} {HH}:{mm}",
+        "Month-Hourly": "{dd} {MMMM} {yyyy} {HH}:{mm}",
+        "Year-Daily": "{dd} {MMMM} {yyyy}",
+        "Last-Day": "{dd} {MMMM} {yyyy} {HH}:{mm}",
+        "Last-Week": "{dd} {MMMM} {yyyy} {HH}:{mm}",
+        "Last-Month": "{dd} {MMMM} {yyyy} {HH}:{mm}",
+        "Last-Year": "{dd} {MMMM} {yyyy}",
+    };
+
+    #monthNames = [
```

Comment:
No doubt that there is a cleaner way to get month names using [ES6 Internationalization API](https://generatedcontent.org/post/59403168016/esintlapi).
See an example here: https://codingbeautydev.com/blog/javascript-convert-month-number-to-name/

Anyway it is enough for now. We will improve it later.